django/contrib/admin/views/autocomplete.py

```diff
@@ -41,6 +74,7 @@ def get_paginator(self, *args, **kwargs):
     def get_queryset(self):
         """Return queryset based on ModelAdmin.get_search_results()."""
         qs = self.model_admin.get_queryset(self.request)
+        qs = qs.complex_filter(self.source_field.get_limit_choices_to())
```

Comment:
As far as I'm aware, this can create duplicates on multi-value relations, see related fix for forms 556fa4bbba5ba86bc1646a86fb11ab55405d4aa4.

Reply:
Yes, it does. This is handled two lines below.

Reply:
It looks like this can still produce duplicates, see [comment](https://code.djangoproject.com/ticket/15819#comment:16). But that's a separate issue.

Reply:
Hm… I see, but yes, that is a separate issue.